packages/zowe-explorer/src/trees/dataset/DatasetTree.ts

```diff
@@ -113,136 +112,278 @@ export class DatasetTree extends ZoweTreeProvider<IZoweDatasetTreeNode> implemen
+            );
 
             if (!recursiveCall) {
-                // Delete any files from the selection on the source LPAR
                 await vscode.workspace.fs.delete(sourceNode.resourceUri, { recursive: false });
-                //dialog message for successful move of pds and members
-                Gui.infoMessage(vscode.l10n.t("Data set(s) moved successfully."));
+                Gui.infoMessage(vscode.l10n.t("Data moved successfully."));
+            }
+        } catch (err) {
+            if (err instanceof Error) {
+                Gui.errorMessage(vscode.l10n.t("Failed to move {0}: {1}", dsname, err.message));
             }
         }
     }
 
     public async handleDrop(
         targetNode: IZoweDatasetTreeNode | undefined,
         dataTransfer: vscode.DataTransfer,
-        _token: vscode.CancellationToken
+        _token?: vscode.CancellationToken
     ): Promise<void> {
         const droppedItems = dataTransfer.get("application/vnd.code.tree.zowe.ds.explorer");
         if (!droppedItems) return;
 
         let target = targetNode;
         if (!target) return;
 
-        // check each dropped node for same-object, member collision, and structure issues
+        // Pivot ONLY when dropping onto a PDS child (member / placeholder), so the PDS is the real container.
+        if (!SharedContext.isPds(target)) {
+            const parent = target.getParent?.();
+            const isMemberLike =
+                SharedContext.isDsMember(target) || target.contextValue === Constants.DS_MEMBER_CONTEXT;
+
+            if (parent && SharedContext.isPds(parent) && isMemberLike) {
+                target = parent as IZoweDatasetTreeNode;
+            }
+        }
+
+        // Validate each dropped node: same-object, member collision, and structure issues
         for (const item of droppedItems.value) {
-            const node = this.draggedNodes[item.uri.path];
+            const node = this.draggedNodes[item.uri.path] as IZoweDatasetTreeNode | undefined;
             if (!node) continue;
+
             const srcDsn = SharedUtils.getNodeProperty(node, "label");
 
-            // 1. Same-object check
-            if (node.getProfile && target.getProfile && (await SharedUtils.isSamePhysicalDataset(node.getProfile(), target.getProfile(), srcDsn))) {
-                Gui.errorMessage(vscode.l10n.t(SharedUtils.ERROR_SAME_OBJECT_DROP));
+            // 0) Sequential datasets are NEVER valid drop targets (so seq->seq is blocked too)
+            if (SharedContext.isDs(target)) {
+                Gui.errorMessage(vscode.l10n.t("Cannot drop items into a sequential dataset."));
                 return;
             }
 
-            // 2. PDS member collision check (only if both are PDS)
+            // 1) Same-object check
+            const nodeProfile = node.getProfile?.();
+            const targetProfile = target.getProfile?.();
+            const nodeParentUri = node.getParent?.()?.resourceUri?.path;
```

Comment:
Are there cases where the `getProfile` is `undefined`? The optional chaining seems redundant here.